setup.py

```diff
@@ -17,22 +17,54 @@
 import sys
 import subprocess
 import shutil
+import json
 from pathlib import Path
 from setuptools import setup, find_packages
 
+def envar():
```

Comment:
Just some thoughts about this approach for the backend information.
This JSON approach is reasonable but is quite Python-forward. This may get in the way of implementing a C++-only device. This is not a blocking condition but would be cool to keep the path to a full C++ device as much clean as possible. Did you try to do this in de C++ layer? Maybe with the use of flags like [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/cmake/process_options.cmake#L246).

Reply:
The initial plan discussed with Lee is to allow users to query the build info from the python layer.  Do you mean we can also add query support for our cpp lib? I'm not sure if I got your point.

Reply:
We could input this information in the C++ layer, in a struct, and write a binding for it to take it to the python layer, like [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/src/bindings/Bindings.hpp#L338-L342). Maybe there is even a way to query Kokkos for the device and architecture information.

Reply:
Thanks for the clarification. Let me check.